enable/trait_defs/ui/editor_with_component.py

```diff
@@ -0,0 +1,133 @@
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from traits.api import Bool, Instance, Str, observe
+from traitsui.api import BasicEditorFactory, Editor as BaseEditor, toolkit_object
+
+from enable.component import Component
+from enable.label import Label
+from enable.window import Window
+from enable.enable_traits import font_trait
+
+
+Editor = toolkit_object("editor:Editor")
+if not issubclass(Editor, BaseEditor):
+    Editor = object
+
+
+class EditorWithComponent(Editor):
+    """Base class for editors which hold a Component that displays the value.
+
+    This is distinct from CompononentEditor in that the value is not the
+    Component.
+    """
+
+    #: The window that displays the component.
+    window = Instance(Window)
+
+    #: The component that is created by the UI.
+    component = Instance(Component)
+
+    def init(self, parent):
+        self.component = self.create_component()
+        size = self._get_initial_size()
+        self.window = self.create_window(parent, size)
+        self.control = self.window.control
+        self._parent = parent
+
+    def create_component(self):
+        raise NotImplementedError()
+
+    def create_window(self, parent, size):
+        window = Window(
+            parent,
+            size=size,
+            component=self.component,
+            high_resolution=self.factory.high_resolution,
+            bgcolor='sys_window',
+        )
+        return window
+
+    def dispose(self):
+        if self.window is not None:
+            self.window.cleanup()
+            self.component = None
+            self.window = None
+            self._parent = None
+        super().dispose()
+
+
+class EditorWithLabelComponent(EditorWithComponent):
+    """A class that creates a Label component.
+
+    By default it displays the string representation of the value.
+    """
+
+    #: The font to use for the label.
+    font = font_trait()
+
+    def create_component(self):
+        """Creates the label component."""
+        component = Label(
+            hjustify="center",
+            vjustify="center",
+            resizable='hv',
+            text=self.str_value,
+            font=self.font,
+        )
+        return component
+
+    def update_editor(self):
+        if self.component is not None:
+            self.component.text = self.str_value
+            self.component.invalidate_and_redraw()
+
+    @observe('font')
+    def update_font(self, event):
+        if self.component is not None:
+            self.component.font = self.font
+            self.component.invalidate_and_redraw()
+
+    def _get_initial_size(self):
+        width = self.item.width
+        height = self.item.height
+
+        if width < 0:
+            width = 200
+        if height < 0:
+            height = 50
+
+        return width, height
+
+    def set_size_policy(self, direction, resizable, springy, stretch):
```

Comment:
It would be nice to have some documentation about the expected types of the various arguments here.

Reply:
This is an internal Qt backend method (despite being public): full documentation is here https://github.com/enthought/traitsui/blob/44459d729ca013a6b58cebda2bf94bd789de8b7c/traitsui/qt4/editor.py#L298-L322

I can copy it over if you think it is needed.